### PR TITLE
planner: Introduce a new global variable to control the historical statistics feature

### PR DESCRIPTION
**Pull request by An-DJ**
### What problem does this PR solve?

Issue Number:  #18745 

Problem Summary: Introduce a new global variable as the switch to control the behavior

### What is changed and how it works?
Use `tidb_enable_historical_stats` to control the behavior above.

Default is OFF.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: sessionctx/variable/tidb_vars.go [hunk 1]
```diff
@@ -620,6 +620,8 @@ const (
 	TiDBGCScanLockMode = "tidb_gc_scan_lock_mode"
 	// TiDBEnableEnhancedSecurity restricts SUPER users from certain operations.
 	TiDBEnableEnhancedSecurity = "tidb_enable_enhanced_security"
+	// TiDBEnableHistoricalStats enables the historical statistics feature (default off)
+	TiDBEnableHistoricalStats = "tidb_enable_historical_stats"
```
Comment (qw4990) [inline sessionctx/variable/tidb_vars.go:624]: Please add some tests for it. You can add them here [set_test.go#L43](https://github.com/pingcap/tidb/blob/master/executor/set_test.go#L43) .

**Review by qw4990 — APPROVED** [review]

**Review by rebelice — APPROVED** [review]

**Comment by morgo** [comment]
I caught this a little late, but I have a small improvement I'd like to make here: https://github.com/pingcap/tidb/issues/30821 -- not your fault; we can make it clearer these internal APIs are not meant to be used any more.

**Pull request merged**